plugin/notes/notes.html

```diff
@@ -574,6 +575,11 @@ <h4 class="label">Notes</h4>
 								// And now we replace every zero-value timing with that average
 								timings = timings.map( function(x) { return (x==0 ? timePerSlide : x) } );
 							}
+							var slidesUnderMinimum = timings.filter( function(x) { return (x < minTimePerSlide) } ).length
+							if ( slidesUnderMinimum ) {
+								message = "The calculated pacing time for " + slidesUnderMinimum + " slide(s) is under the configured minimum of " + minTimePerSlide + " seconds. Check the data-timing attribute on individual slides, or consider increasing the totalTime or minimumTimePerSlide configuration options (or removing some slides).";
```

Comment:
> Either way, the important thing is to phrase the warning in a way which makes it clear how to fix the issue.

@aspiers: Would you think that this is clear enough for the alert message?

Reply:
I suggest this small tweak:

```suggestion
								message = "The calculated pacing time for " + slidesUnderMinimum + " slide(s) without the data-timing attribute is under the configured minimum of " + minTimePerSlide + " seconds. Check the data-timing attribute on other slides, or consider increasing the totalTime or minimumTimePerSlide configuration options (or removing some slides).";
```

Reply:
That tweak is at odds with what the code does, though. :)

The preceding test checks _all_ slides for whether they're under the minimum, including the ones that have `data-timing` set. I was on the fence about what to do here, but figured that if you do set a nonzero minimum time per slide, then you'd also want to be warned when a slide _with_ `data-timing` set the pacing time (presumably unintentionally) below the intended minimum. Say your minimum is 5, and you meant to set `data-timing` on a slide to 30 but typoed it to 3, I think it's not unreasonable to assume you'd like to be warned about that.

Reply:
Ah, good point. How about making it more explicit so that it explains whether slides with the attribute are below the limit, or slides without are, or some of both?

Reply:
In the interest of future contributors/maintainers, I'd prefer to keep this change minimal, and forgo that idea.

I'm really not trying to sound lazy here, but unless I'm mistaken then for your suggestion to work it'd take a separate counter for `data-timing` enabled slides. And I think that would just make this bit more convoluted (read: less maintainable), for little tangible benefit to the user. If a presenter is using the pacing timer — which you can only enable by editing the configuration —, chances are that they're well capable of running a recursive grep for the strings `dataTiming`, `totalTime`, and `minimumTimePerSlide`.

This stance might sound ridiculous for a tiny change like that, but those changes do add up — and if I can make Hakim's life just a wee bit easier without harming UX, I'll take that opportunity. :)

Reply:
Sounds good to me! You are doing the right thing by favouring simplicity; I have a tendency to over-engineer things 😜

Reply:
Excellent, glad we agree. Would you say that I can go ahead and propose the PR in its current state? If so, then please resolve this conversation and add your +1 on the review — otherwise, please send me any other nits you want me to pinch. ;)

Reply:
Seems I don't have enough permissions here for a Resolve Conversation button, but I've approved the PR.

Reply:
Huh, interesting. OK then I'll go ahead and do that. :)